src/plugins/discover/public/application/main/hooks/use_filters_validation.ts

```diff
@@ -1,56 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-import { IToasts, ToastsStart } from '@kbn/core/public';
-import { FilterManager } from '@kbn/data-plugin/public';
-import { i18n } from '@kbn/i18n';
-import { SavedSearch } from '@kbn/saved-search-plugin/public';
-import { useEffect } from 'react';
-import { debounceTime } from 'rxjs';
-
-const addInvalidFiltersWarn = (toastNotifications: IToasts) => {
```

Comment:
I think we should remove or rework it, because notifying users about the data view filter references being different because of the change of the data view id when the data view is temporary and the id has changed because the user has edited the data view, which caused an id change since it's temporary, I don't think it's helpful.... but it should be worked on in a different PR, so now I adapted it instead of removing. To narrow the scope a bit.
<img width="905" alt="Bildschirmfoto 2023-04-28 um 11 51 48" src="https://user-images.githubusercontent.com/463851/235144389-14652cb5-82ef-45a5-9b25-91d583011178.png">